R/install.snowball.R

```diff
@@ -72,69 +72,130 @@ install.snowball <- function(pkg, date, include.suggests, force.install = FALSE,
+            date <- as.Date(date) # Ensure formated as date
+            max.date <- min(Sys.Date() - 2, date + 10) # use as highest possible mran date, 2 days ago
+            mran.date <- get.available.mran.date(date - 10, max.date) # Function get.available.mran.date() in utils.R
+
+            # Try path for current packages
+            url3 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url3, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+          }
+        # Attempt 4 overall, MRAN around selected date, Archive tarballs path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            url4 <- paste0("https://cran.microsoft.com/snapshot/", mran.date, "/src/contrib/Archive/", pkg.k, "/", pkg_vrs.k, ".tar.gz")
+            install.packages(url4, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+          }
+
+        # Attempt 5, MRAN around Publish date, current path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            # Set date for mran, midpoint of available dates around requested date  +-10 days,
+            toc.pkg <- toc(pkg)
+            date.Published <- as.Date(subset(toc.pkg, Version = vrs)$Published) + 2
+            max.date.published <- min(Sys.Date() - 2, date.Published + 20) # Use as highest possible mran published date, 2 days ago
+            mran.published.date <- get.available.mran.date(date.Published, max.date.published) # Function get.available.mran.date() in utils.R
+            # Treat as current
+            url5 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url5, repos = NULL, lib = lib.k, type = "source", dependencies = F, quiet = quiet.install)
+          }
+
+        # Attempt 6, MRAN around Publish date, Archive path
+        if (is.pkg_vrs.installed(pkg.k, vrs.k) == FALSE) # if still not installed
+          {
+            # If fails try archive
+            url6 <- paste0("https://cran.microsoft.com/snapshot/", mran.published.date, "/src/contrib/", pkg_vrs.k, ".tar.gz")
+            install.packages(url6, repos = NULL, lib = lib.k, type = "source", quiet.install = quiet, dependencies = F, quiet = quiet.install)
+          }
       }
+      # install.source(pkg_vrs.k, lib.k, date, quiet.install = quiet.install)  #used to call an external function
+
 
       # 3.8 VERIFY INSTALL
       now.installed <- is.pkg_vrs.installed(pkg.k, vrs.k)
 
       # 3.8.5 If not success, try source again, forcing download of file
-      if (!now.installed) {
-        install.source(pkg_vrs.k, lib.k, date, force.download = TRUE, quiet.install = quiet.install)
-      }
+      # if (!now.installed) {
```

Comment:
It's better to remove code rather than commenting it out. We have the history since it's version controlled (with git). Commented code often go stale and nobody remembers why it was here, if it's safe to remove it, etc.